plugins/woocommerce/tests/php/src/Blocks/BlockTypes/BlockifiedProductDetails.php

```diff
@@ -0,0 +1,350 @@
+<?php
+
+declare(strict_types=1);
+namespace Automattic\WooCommerce\Tests\Blocks\BlockTypes;
+
+use WC_Helper_Product;
+
+/**
+ * Tests for the BlockifiedProductDetails block type
+ */
+class BlockifiedProductDetails extends \WP_UnitTestCase {
+
+	/**
+	 * Page ID
+	 *
+	 * @var @string
+	 */
+	private static $page_id;
+
+	/**
+	 * Product
+	 *
+	 * @var @WC_Product
+	 */
+	private static $product;
+
+	/**
+	 * Create Simple Product and Page
+	 */
+	public static function setUpBeforeClass(): void {
+		parent::setUpBeforeClass();
+
+		self::$product = WC_Helper_Product::create_simple_product( false );
+		WC_Helper_Product::create_product_review( self::$product );
+
+		self::$page_id = wp_insert_post(
+			[
+				'post_title'  => 'Test Product Page',
+				'post_type'   => 'page',
+				'post_status' => 'publish',
+			],
+			true
+		);
+	}
+	/**
+	 * Set up product and page for each test
+	 *
+	 * @return void
+	 */
+	public function setUp(): void {
+		parent::setUp();
+		global $post, $product;
+
+		$post = get_post( self::$page_id );
+		setup_postdata( $post );
+		$product            = self::$product;
+		$GLOBALS['product'] = $product;
+	}
+
+	/**
+	 * Reset postdata after each test
+	 *
+	 * @return void
+	 */
+	public function tearDown(): void {
+		parent::tearDown();
+		wp_reset_postdata();
+	}
+
+	/**
+	 * Delete the product and page after all tests
+	 *
+	 * @return void
+	 */
+	public static function tearDownAfterClass(): void {
+		parent::tearDownAfterClass();
+		wp_delete_post( self::$page_id, true );
+		WC_Helper_Product::delete_product( self::$product->get_id() );
+	}
+
+
+	/**
+	 * Test Product Details render function when `woocommerce_product_tabs` hook isn't used
+	 * IMPORTANT: The current test doesn't validate the entire HTML, but only the text content inside the HTML.
+	 * This is because some ids are generated dynamically via wp_unique_id that it is not straightforward to mock.
+	 */
+	public function test_product_details_render_with_no_hook() {
+
+		$template = '<!-- wp:woocommerce/blockified-product-details --><div class="wp-block-woocommerce-blockified-product-details"><!-- wp:woocommerce/accordion-group {"autoclose": false} --><div class="wp-block-woocommerce-accordion-group"><!-- wp:woocommerce/accordion-item {"openByDefault": false} --><div class="wp-block-woocommerce-accordion-item"><!-- wp:woocommerce/accordion-header --><h3 class="wp-block-woocommerce-accordion-header accordion-item__heading"><button class="accordion-item__toggle"><span>Description Header</span><span class="accordion-item__toggle-icon has-icon-plus" style="width:1.2em;height:1.2em"><svg width="1.2em" height="1.2em" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg" aria-hidden="true"><path d="M11 12.5V17.5H12.5V12.5H17.5V11H12.5V6H11V11H6V12.5H11Z" fill="currentColor"></path></svg></span></button></h3>
```

Comment:
Do you know if there is a nicer way to handle this?
Having to write out the mark up like this makes the test really hard to read.

Reply:
Moving this to an `html` file would also help

Reply:
Or could we format it a bit better like this: https://github.com/WordPress/gutenberg/blob/b05bee7e2e8eefb2be06c228002e4f9783847747/phpunit/blocks/render-post-template-test.php#L53

Reply:
Even using the Gutenberg approach, the editor can't format the HTML. For this reason, I moved the HTML in a dedicated folder with 69a736f386d27dd5ecae2177ef98c2b01064823e.